tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/Emitter.scala

```diff
@@ -66,32 +79,72 @@ final class Emitter private (semantics: Semantics, outputMode: OutputMode,
     emitLines(customHeader, builder)
 
   def emitPrelude(builder: JSFileBuilder, logger: Logger): Unit = {
-    outputMode match {
-      case OutputMode.ECMAScript51Global =>
-      case OutputMode.ECMAScript51Isolated | OutputMode.ECMAScript6 =>
-        builder.addLine("(function(){")
+    moduleKind match {
+      case ModuleKind.NoModule =>
+        outputMode match {
+          case OutputMode.ECMAScript51Global =>
+          case OutputMode.ECMAScript51Isolated | OutputMode.ECMAScript6 =>
+            builder.addLine("(function(){")
+        }
+
+      case ModuleKind.CommonJSModule =>
     }
 
     builder.addLine("'use strict';")
-    builder.addFile(CoreJSLibs.lib(semantics, outputMode))
+    builder.addFile(CoreJSLibs.lib(semantics, outputMode, moduleKind))
   }
 
   def emit(unit: LinkingUnit, builder: JSTreeBuilder, logger: Logger): Unit = {
     startRun(unit)
     try {
       val orderedClasses = unit.classDefs.sortWith(compareClasses)
+
+      emitModuleImports(orderedClasses, builder)
+
       for (classInfo <- orderedClasses)
         emitLinkedClass(classInfo, builder)
     } finally {
       endRun(logger)
     }
   }
 
+  private def emitModuleImports(orderedClasses: List[LinkedClass],
```

Comment:
I don't think so. `ClassEmitter` is about class-level things. This is linking-unit-level. So IMO it is similar to emitting the prelude/postlude (including the CoreJSLib).